generic3g/actions/AccumulatorAction.F90

```diff
@@ -62,23 +63,45 @@ subroutine initialize(this, importState, exportState, clock, rc)
 
       integer :: status
       type(ESMF_Field) :: import_field, export_field
-      logical :: fields_are_conformable
+      type(ESMF_TypeKind_Flag) :: typekind
+      logical :: conformable
+      logical :: same_typekind
 
+      conformable = .FALSE.
+      same_typekind = .FALSE.
       call get_field(importState, import_field, _RC)
+      call ESMF_FieldGet(import_field, typekind=typekind, _RC)
+      _ASSERT(typekind==ESMF_TYPEKIND_R4, 'Only ESMF_TYPEKIND_R4 is supported.')
       call get_field(exportState, export_field, _RC)
+      conformable = FieldsAreConformable(import_field, export_field, _RC)
+      _ASSERT(conformable, 'Import and export fields are not conformable.')
+      same_typekind = FieldsAreSameTypeKind(import_field, export_field, _RC)
+      _ASSERT(same_typekind, 'Import and export fields are different typekinds.')
+      this%typekind = typekind
```

Comment:
I think some (well-chosen) empty lines would improve readability in this procedure.   I would not bother but since there are changes to use SELECT CASE above, might as well do it.